src/python/pants/backend/python/tasks/python_binary_create.py

```diff
@@ -128,6 +128,15 @@ def _create_binary(self, binary_tgt, results_dir):
       # Dump everything into the builder's chroot.
       for tgt in source_tgts:
         dump_sources(builder, tgt, self.context.log)
+      # We need to ensure that we are resolving for only the current platform if we are
+      # including local python dist targets that have native extensions.
+      if self.tgt_closure_has_native_sources():
+        platforms = self.tgt_closure_platforms()
+        if platforms != ['current']:
+          raise IncompatiblePlatformsError('The target set contains one or more targets that depend on '
```

Comment:
This would be much more useful if it listed (the first two?) incompatible targets: ... you could maybe make `self.tgt_closure_platforms()` return a `dict(str->list)` of targets by platform, and then render the first two incompatible targets.